Source/WebCore/platform/mediastream/mac/ScreenCaptureKitCaptureSource.mm

```diff
@@ -116,11 +116,39 @@ - (void)stream:(SCStream *)stream didStopWithError:(NSError *)error
 
 - (void)stream:(SCStream *)stream didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer ofType:(SCStreamOutputType)type
 {
+    ASSERT(type == SCStreamOutputTypeScreen);
+
+    if (!sampleBuffer)
+        return;
+
+    auto attachments = (__bridge NSArray *)PAL::CMSampleBufferGetSampleAttachmentsArray(sampleBuffer, false);
+    SCFrameStatus status = SCFrameStatusStopped;
+    [attachments enumerateObjectsUsingBlock:makeBlockPtr([&] (NSDictionary *attachment, NSUInteger, BOOL *stop) {
+        auto statusNumber = (NSNumber *)attachment[SCStreamFrameInfoStatus];
+        if (!statusNumber)
+            return;
+
+        status = (SCFrameStatus)[statusNumber integerValue];
+        *stop = YES;
+    }).get()];
+
+    switch (status) {
+    case SCFrameStatusStarted:
+    case SCFrameStatusComplete:
```

Comment:
Should we proceed if SCFrameStatusStarted?

Reply:
Yes, `SCFrameStatusStarted` signals the first frame sent when a stream starts.